shodan/exception.py

```diff
@@ -1,7 +1,8 @@
 class APIError(Exception):
     """This exception gets raised whenever a non-200 status code was returned by the Shodan API."""
-    def __init__(self, value):
+    def __init__(self, value, status_code=None):
```

Comment:
Since the project isn't using type hinting, could we set the default value of `status_code` to 0 or -1, rather than None? 

I'm just thinking that this would provide for a more future proof solution and would help prevent any future TypeErrors if future enhancements ever needed to perform comparisons on status codes.

Using 0 or -1 as default values also shouldn't cause any confusion given that these aren't existing [HTTP Status codes](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status).

Reply:
Yeah totally get your point. I changed the default to be 0 which is nice one for "no status code" in my opinion.

Reply:
Looks good!